src/measure_theory/decomposition/signed_hahn.lean

```diff
@@ -0,0 +1,644 @@
+/-
+Copyright (c) 2021 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import measure_theory.vector_measure
+import order.symm_diff
+
+/-!
+# Unsigned Hahn decomposition
+
+This file defines positive and negative sets with respect to a vector measure and
+prove the Hahn decomposition (signed version).
+
+## Main definitions
+
+* `vector_measure.positive` : a set `i` is positive with respect to a vector measure if for all
+  measurable sets `j` contained in `i`, the measure of `j` is non-negative.
+* `vector_measure.negative` : a set `i` is negative with respect to a vector measure if for all
+  measurable sets `j` contained in `i`, the measure of `j` is non-positive.
+
+## Main results
+
+* `signed_measure.exists_disjoint_positive_negative_union_eq` : the Hahn decomposition theorem.
+* `signed_measure.exists_negative_set` : A measurable set of negative measure contains at least
+  one negative subset.
+
+## Tags
+
+positive, negative, Hahn decomposition
+-/
+
+noncomputable theory
+open_locale classical big_operators nnreal ennreal
+
+section lemmas
+-- ↓ I will move the content of this section to the approprate files after the reviews so we don't
+-- need to compile all of mathlib everytime.
+
+variables {α : Type*}
+
+open set filter
+
+lemma set.Union_inter_diff_eq {f : ℕ → set α} {a : set α} (ha : a ⊆ ⋃ n, f n) :
```

Comment:
Is this and the next lemma useful enough to warrant its existence? Should I just make it a private lemma where I use it?

Reply:
See my comment below. I think there's an interesting lemma hiding in this proof, but I can't work out how to generalize it.